OutOfSchool/OutOfSchool.WebApi.Tests/Services/NotificationServiceTests.cs

```diff
@@ -0,0 +1,270 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Moq;
+using NUnit.Framework;
+using OutOfSchool.Services.Enums;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+using OutOfSchool.Tests.Common;
+using OutOfSchool.WebApi.Config;
+using OutOfSchool.WebApi.Hubs;
+using OutOfSchool.WebApi.Models.Notifications;
+using OutOfSchool.WebApi.Services;
+using OutOfSchool.WebApi.Util;
+using OutOfSchool.WebApi.Util.Mapping;
+
+namespace OutOfSchool.WebApi.Tests.Services;
+
+[TestFixture]
+public class NotificationServiceTests
+{
+    private INotificationService notificationService;
+    private Mock<INotificationRepository> notificationRepositoryMock;
+    private Mock<ILogger<NotificationService>> loggerMock;
+    private Mock<IStringLocalizer<SharedResource>> localizerMock;
+    private IMapper mapper;
+    private Mock<IHubContext<NotificationHub>> notificationHubMock;
+    private Mock<IOptions<NotificationsConfig>> notificationsConfigMock;
+
+    private string userId;
+    private List<Notification> notifications;
+
+    [SetUp]
+    public void SetUp()
+    {
+        notificationRepositoryMock = new Mock<INotificationRepository>();
+        loggerMock = new Mock<ILogger<NotificationService>>();
+        localizerMock = new Mock<IStringLocalizer<SharedResource>>();
+        mapper = TestHelper.CreateMapperInstanceOfProfileTypes<CommonProfile, MappingProfile>();
+        notificationHubMock = new Mock<IHubContext<NotificationHub>>();
+        notificationsConfigMock = new Mock<IOptions<NotificationsConfig>>();
+
+        notificationService = new NotificationService(
+            notificationRepositoryMock.Object,
+            loggerMock.Object,
+            localizerMock.Object,
+            mapper,
+            notificationHubMock.Object,
+            notificationsConfigMock.Object);
+
+        userId = Guid.NewGuid().ToString();
+        notifications = CreateNotificationsForUser(userId);
+    }
+
+    #region ReadAll
+    [Test]
+    public async Task ReadAll_WhenUserIdIsValid_ShouldSetReadDateTimeForAllUnreaded()
+    {
+        // Arrange
+        notificationRepositoryMock
+            .Setup(n => n.SetReadDateTimeForAllUnreaded(It.IsAny<string>(), It.IsAny<DateTimeOffset>()))
+            .Returns(Task.CompletedTask);
+
+        // Act
+        await notificationService.ReadAll(userId).ConfigureAwait(false);
+
+        // Assert
+        notificationRepositoryMock
+            .Verify(
+                x => x.SetReadDateTimeForAllUnreaded(
+                userId,
+                It.IsAny<DateTimeOffset>()),
+                Times.Once());
```

Comment:
(Test code style) You could simplify this test code (and remove duplications) using expected `userId` value (instead of `It.IsAny<string>()`) in `mock.Setup()`, and using parameterless `mock.VerifyAll()` (instead of specific `.Verify()`). 

Interesting fact is that you've used this approach in the subsequent tests, but not here.

Reply:
> (Test code style) You could simplify this test code (and remove duplications) using expected `userId` value (instead of `It.IsAny<string>()`) in `mock.Setup()`, and using parameterless `mock.VerifyAll()` (instead of specific `.Verify()`).
> 
> Interesting fact is that you've used this approach in the subsequent tests, but not here.

I used `VerifyAll` in this test but removed it before committing. Thank you!